docs/2.10.0/docs/concepts/ledger-model/upgrade-model-in-depth.rst

```diff
@@ -279,22 +279,55 @@ template on the left because it changes the type of ``x1`` from ``Int`` to ``Tex
 Template Keys
 ~~~~~~~~~~~~~
 
-The new version of a template cannot modify the key of the prior version of a
-template in any way.
+The key of the new version of a template must be a valid upgrade of the key of
+the prior version of that template.
 
-Adding a key leads to a validation error.
 
-Removing a key leads to a validation error.
-
-Changing the type of a key leads to a validation error.
-
-For Daml 2.9, key types can only use definitions from the current package or 
-from the Daml standard library. 
+Adding a key, removing a key, or changing the type of the key for a non-valid
+upgrade leads to a validation error.
 
 .. _examples-3:
 
 **Examples**
 
+Below, the template on the right is a valid upgrade of the template on the left:
+the type of the key of the template on the right is a valid upgrade of the type
```

Comment:
```suggestion
the type of the key of the template on the right is a valid upgrade of the type
```
```suggestion
this is because the type of the key of the template on the right is a valid upgrade of the type
```